Wrappers/Python/cil/optimisation/functions/Function.py

```diff
@@ -320,7 +326,88 @@ def __add__(self, other):
         else:
             return super(SumFunction, self).__add__(other)
 
+class SubsetSumFunction(SumFunction):
+
+    r"""SubsetSumFunction represents the following sum 
+    
+    .. math:: \frac{1}{n}\sum_{i=1}^{n} F_{i} = \frac{1}{n}(F_{1} + F_{2} + ... + F_{n})
+
+    where :math:`n` is the number of subsets.
+
+    Parameters:
+    -----------
+    functions : list(functions) 
+                A list of functions: :code:`[F_{1}, F_{2}, ..., F_{n}]`. Each function is assumed to be smooth function with an implemented :func:`~Function.gradient` method.
+    sampling : :obj:`string`, Default = :code:`random`
+               Selection process for each function in the list. It can be :code:`random` or :code:`sequential`. 
+    replacement : :obj:`boolean`. Default = :code:`True`
+               The same subset can be selected when :code:`replacement=True`. 
+                
+    Example
+    -------
+
+    .. math:: \sum_{i=1}^{n} F_{i}(x) = \sum_{i=1}^{n}\|A_{i} x - b_{i}\|^{2}
+
+    >>> f = SubsetSumFunction([LeastSquares(Ai, b=bi)]*n for Ai,bi in zip(A_subsets, b_subsets))   
+  
+    """
+    
+    def __init__(self, functions, 
+                 sampling = "random",
+                 replacement = True,
+                 subset_init = -1):    
+        
+        super(SubsetSumFunction, self).__init__(*functions)
+        
+        self.subset_num = subset_init
+        self.data_passes = [0]
+        self.sampling = sampling  
```

Comment:
Might be useful to also track the used subset_num used per iteration, by say creating an attribute self.subset_use = [subset_init], and then appending to it in each iteration

Reply:
The current implementation does not seem to track the `subset_use`. One can create an attribute `self.subset_use` here, and then append to it after calling `next_subset` with a given function (or appending to it within the `next_subset` method, but then making sure that svrg does not call `next_subset` directly after it has updated the snapshots)